ResumableUpload.CS.Sample/ResumeableUploadSample.cs

```diff
@@ -0,0 +1,241 @@
+                        });
+                    //
+                    // Default Timeout is 1 minute 40 seconds. 
+                    // The following changes it to 2 minutes. 
+                    objYouTubeService.HttpClient.Timeout = TimeSpan.FromMinutes(HTTP_CLIENT_TIMEOUT_MINUTES);
+                    bOKtoContinue = true;
+                    }
+                catch (Exception ex)
+                    {
+                    Console.WriteLine(String.Format("YouTubeService error: {0}", ex.Message));
+                    }
+
+                if (bOKtoContinue)
+                    {
+                    Video objVideo  = new Video();
+                    VideoSnippet vs = new VideoSnippet
+                    {
+                        Title = "Test Title",
+                        Description = "Test Description",
+                        Tags = new string[] { "keyword1", "keyword2", "keyword3" },
+                        CategoryId = "25" // News & Politics
+                    };
+                    objVideo.Snippet = vs;
+                    objVideo.Status = new VideoStatus {Embeddable = true,PrivacyStatus = "private"};
+                    objVideo.RecordingDetails = new VideoRecordingDetails 
+                                        {Location = new GeoPoint 
+                                            {Latitude = 41.327464, Longitude = -72.194555, Altitude = 10},
+                                            LocationDescription = "East Lyme, CT, U.S.A.",
+                                            RecordingDate=DateTime.Now.Date};
+                    string strFilePath = "C:\\Users\\YourUserName\\Videos\\Video\\YourVideo.mp4";
+                    Globals.FileLength = new FileInfo(strFilePath).Length;
+                    Globals.ProgressPercent = 0;
+                    using (FileStream objFileStream = new FileStream(strFilePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                        {
+                        objVideosInsertRequest = objYouTubeService.Videos.Insert(objVideo, "snippet,status,recordingDetails", objFileStream, "video/*");
+                        objVideosInsertRequest.ChunkSize = intHalfMBChunkSize;  // .5MB
+                        objVideosInsertRequest.ResponseReceived += videosInsertRequest_ResponseReceived; // Add event handler
+                        objVideosInsertRequest.ProgressChanged += videosInsertRequest_ProgressChanged;   // Add event handler
+                        objCancellationToken = new System.Threading.CancellationToken();
+                        Globals.ResumeRetries = 0;
+                        try
+                        {
+                            await objVideosInsertRequest.UploadAsync(objCancellationToken);
+                        }
+                        catch
+                        {
+                            // Error handling is done in videosInsertRequest_ProgressChanged() Event
```

Comment:
Do we at least want to write something to the console? Or would that be redundant?